renderdoc/driver/metal/metal_types.h

```diff
@@ -24,16 +24,30 @@
 
 #pragma once
 
-#include "metal_common.h"
+#include "api/replay/rdcstr.h"
+#include "official/metal-cpp.h"
+#include "serialise/serialiser.h"
 
 #define METALCPP_WRAPPED_PROTOCOLS(FUNC) \
   FUNC(Device);                          \
   FUNC(Function);                        \
   FUNC(Library);
 
-#define DECLARE_OBJC_HELPERS(CPPTYPE) \
-  class WrappedMTL##CPPTYPE;          \
-  extern MTL::CPPTYPE *AllocateObjCWrapper(WrappedMTL##CPPTYPE *wrapped);
+#define DECLARE_OBJC_HELPERS(CPPTYPE)                                \
+  class WrappedMTL##CPPTYPE;                                         \
+  extern WrappedMTL##CPPTYPE *GetWrapped(MTL::CPPTYPE *objCWrapped); \
+  extern MTL::CPPTYPE *GetReal(MTL::CPPTYPE *objCWrapped);           \
+  extern bool IsObjCWrapped(MTL::CPPTYPE *objCWrapped);              \
+  extern ResourceId GetId(MTL::CPPTYPE *objCWrapped);                \
```

Comment:
Minor nitpick, this free function should be `GetResID` to match the other backends and be consistent. There's already a `GetResID` for the generic `WrappedMTLObject`, this would just be an overload which works on the unwrapped type to look up via a cast/GetWrapped call.